IINACT/Windows/MainWindow.cs

```diff
@@ -57,19 +57,11 @@ public override void Draw()
         }
 
         var selectedOverlay = OverlayPresets?[selectedOverlayIndex];
-        var overlayURL = selectedOverlay?.HttpUrl ?? "";
-        if (!string.IsNullOrEmpty(overlayURL))
-        {
-            if (selectedOverlay?.Modern ?? false)
-                overlayURL += $"?OVERLAY_WS=ws://{Server?.Address}:{Server?.Port}/ws";
-            else
-                overlayURL += $"?HOST_PORT=ws://{Server?.Address}:{Server?.Port}";
-            if (!string.IsNullOrEmpty(selectedOverlay?.Options))
-                overlayURL += selectedOverlay.Options;
-        }
+        var overlayUri = selectedOverlay.toOverlayUri($"ws://{Server?.Address}:{Server?.Port}/ws");
```

Comment:
this needs to be null safe or it will crash if the GUI is drawn before OP is ready